### PR TITLE
Gate libsecp256k1 update

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

v0.5.0 of the `libsecp256k1` crate introduces a [fix](https://rustsec.org/advisories/RUSTSEC-2021-0076) that can break consensus during upgrade if not feature gated.

#### Summary of Changes

- Reintroduce recently removed secp256k1 program enablement feature gating
- Adapt above gating for reuse
- Add gated switch to strict secp256k1 signature verification

NOTE: There are no test vectors available to pass the old method and fail the new one. I've reached out to upstream to see if they have some.

### CODE REVIEW

File: sdk/program/src/secp256k1_recover.rs [hunk 1]
```diff
@@ -103,7 +103,10 @@ pub fn secp256k1_recover(
             .map_err(|_| Secp256k1RecoverError::InvalidHash)?;
         let recovery_id = libsecp256k1::RecoveryId::parse(recovery_id)
             .map_err(|_| Secp256k1RecoverError::InvalidRecoveryId)?;
-        let signature = libsecp256k1::Signature::parse_standard_slice(signature)
+
```
Comment (jackcmay) [inline sdk/program/src/secp256k1_recover.rs:106]: This is not used internally and we probably want external users to get the fix right away.  Why not leave this as `parse_standard_slice`?

Comment (t-nelson) [reply sdk/program/src/secp256k1_recover.rs:106]: My initial plan was to keep the client-side and program-side consistent,  let the client slip after feature activation, then be punctual about removing the gating and switching the client-side over

**Review by jackcmay — DISMISSED** [review]

File: sdk/program/src/secp256k1_recover.rs (hunk 1 above)
Comment (t-nelson) [reply sdk/program/src/secp256k1_recover.rs]: As discussed, I went ahead and changed the (likely unused today) client-side non-BPF implementation to anticipate the feature activation.  This'll just drive us to activate faster :wink:

**Comment by t-nelson** [comment]
Working on backports while CI churns.  Zero chance mergify gets anywhere with them

**Pull request merged**